saleor/graphql/account/mutations/permission_group.py

```diff
@@ -38,60 +45,153 @@ class Meta:
+                info.context.user, cleaned_input[field]
             )
-            if indexes:
+            if permissions:
                 error_msg = "You can't add permission that you don't have."
+                permission_enums = [PermissionEnum.get(perm) for perm in permissions]
                 cls.update_errors(
                     errors,
                     error_msg,
-                    "permissions",
-                    AccountErrorCode.NO_PERMISSION,
-                    indexes,
+                    field,
+                    PermissionGroupErrorCode.NO_PERMISSION.value,
+                    permission_enums,
+                    error_field,
                 )
-            cleaned_input["permissions"] = get_permissions(cleaned_input["permissions"])
-        if errors:
-            raise ValidationError(errors)
-        return cleaned_input
+            cleaned_input[field] = get_permissions(cleaned_input[field])
 
     @classmethod
-    def update_errors(cls, errors, msg, field, code, indexes):
-        for index in indexes:
-            error = ValidationError(msg, code=code, params={"index": index})
-            errors[field].append(error)
+    def clean_users(
+        cls,
+        errors: Dict[Optional[str], List[ValidationError]],
+        field: str,
+        cleaned_input: dict,
+        error_field: Optional[str] = None,
+    ):
+        if field in cleaned_input:
+            user_pks = cls.get_user_pks(cleaned_input, field)
+            cls.check_if_users_are_staff(errors, field, user_pks, error_field)
+
+            cleaned_input[f"{field}_pks"] = user_pks
 
     @classmethod
-    def handle_typed_errors(cls, errors: list, **extra):
-        typed_errors = [
-            cls._meta.error_type_class(
-                field=e.field,
-                message=e.message,
-                code=code,
-                index=params.get("index") if params else None,
-            )
-            for e, code, params in errors
+    def get_user_pks(cls, cleaned_input: dict, field: str):
+        if field not in cleaned_input:
+            return []
+
+        user_ids: List[str] = cleaned_input[field]
+
+        user_pks = [
+            from_global_id_strict_type(user_id, only_type=User, field="id")
+            for user_id in user_ids
         ]
-        extra.update({cls._meta.error_type_field: typed_errors})
-        return cls(errors=[e[0] for e in errors], **extra)
+
+        return user_pks
+
+    @classmethod
+    def check_if_users_are_staff(
+        cls,
+        errors: Dict[Optional[str], List[ValidationError]],
+        field: str,
+        user_pks: List[str],
+        error_field: Optional[str] = None,
+    ):
+        non_staff_users = list(
+            account_models.User.objects.filter(pk__in=user_pks)
+            .filter(is_staff=False)
+            .values_list("pk", flat=True)
+        )
+        if non_staff_users:
+            ids = [graphene.Node.to_global_id("User", pk) for pk in non_staff_users]
```

Comment:
Really needed?

Reply:
I need users' graphene `ids` to add them to error.

Reply:
But is it really needed to tell which ones?

Reply:
As above, we decided to return IDs on one of our daily.